From b4a163a135136d5cbe20d243e2c8ddfdb0099549 Mon Sep 17 00:00:00 2001
From: Willy Tarreau <w@1wt.eu>
Date: Wed, 2 Apr 2014 20:55:23 +0200
Subject: [PATCH 118/118] CLEANUP: pattern: move array definitions to proto/
 and not types/

All symbol definitions should be in proto/ and not types/.
---
 include/proto/pattern.h | 9 +++++++++
 include/types/pattern.h | 8 --------
 2 files changed, 9 insertions(+), 8 deletions(-)

diff --git a/include/proto/pattern.h b/include/proto/pattern.h
index 2e1c918..fdf9598 100644
--- a/include/proto/pattern.h
+++ b/include/proto/pattern.h
@@ -28,6 +28,15 @@
 #include <common/standard.h>
 #include <types/pattern.h>
 
+/* pattern management function arrays */
+extern char *pat_match_names[PAT_MATCH_NUM];
+extern int (*pat_parse_fcts[PAT_MATCH_NUM])(const char *, struct pattern *, char **);
+extern int (*pat_index_fcts[PAT_MATCH_NUM])(struct pattern_expr *, struct pattern *, char **);
+extern void (*pat_delete_fcts[PAT_MATCH_NUM])(struct pattern_expr *, struct pat_ref_elt *);
+extern void (*pat_prune_fcts[PAT_MATCH_NUM])(struct pattern_expr *);
+extern struct pattern *(*pat_match_fcts[PAT_MATCH_NUM])(struct sample *, struct pattern_expr *, int);
+extern int pat_match_types[PAT_MATCH_NUM];
+
 void pattern_finalize_config(void);
 
 /* return the PAT_MATCH_* index for match name "name", or < 0 if not found */
diff --git a/include/types/pattern.h b/include/types/pattern.h
index fa29274..a20d6d3 100644
--- a/include/types/pattern.h
+++ b/include/types/pattern.h
@@ -216,14 +216,6 @@ struct pattern_head {
 	struct list head; /* This is a list of struct pattern_expr_list. */
 };
 
-extern char *pat_match_names[PAT_MATCH_NUM];
-extern int (*pat_parse_fcts[PAT_MATCH_NUM])(const char *, struct pattern *, char **);
-extern int (*pat_index_fcts[PAT_MATCH_NUM])(struct pattern_expr *, struct pattern *, char **);
-extern void (*pat_delete_fcts[PAT_MATCH_NUM])(struct pattern_expr *, struct pat_ref_elt *);
-extern void (*pat_prune_fcts[PAT_MATCH_NUM])(struct pattern_expr *);
-extern struct pattern *(*pat_match_fcts[PAT_MATCH_NUM])(struct sample *, struct pattern_expr *, int);
-extern int pat_match_types[PAT_MATCH_NUM];
-
 /* This is the root of the list of all pattern_ref avalaibles. */
 extern struct list pattern_reference;
 
-- 
1.8.3.2

